Detectors/MUON/MCH/Calibration/src/BadChannelCalibrationDevice.cxx

```diff
@@ -35,8 +35,8 @@ void BadChannelCalibrationDevice::init(o2::framework::InitContext& ic)
   mLoggingInterval = ic.options().get<int>("logging-interval") * 1000;
 
   mCalibrator = std::make_unique<o2::mch::calibration::BadChannelCalibrator>();
-  mCalibrator->setSlotLength(o2::calibration::INFINITE_TF);
   mCalibrator->setUpdateAtTheEndOfRunOnly();
+  mCalibrator->setSlotLength(o2::calibration::INFINITE_TF);
```

Comment:
here you changed the order the calls to `setUpdateAtTheEndOfRunOnly` and `setSlotLength` : is it on purpose ? in which case would be good to add a comment right there in the code to say so.
If not, just forget this comment ;-)

Reply:
The order should not matter, but you should set 
```
setSlotLength(o2::calibration::INFINITE_TF);
setMaxSlotsDelay(0);
```
to guarantee that the endOfStream call with `tf = INFINITE_TF`) passes the https://github.com/AliceO2Group/AliceO2/blob/b0f70e90f824acf876f63ca28f8ddeffb6b9523b/Detectors/Calibration/include/DetectorsCalibration/TimeSlotCalibration.h#L315-L317 check.

Reply:
No, it was not done on purpose, reverted.

Reply:
@aferrero2707 a typo slipped to my previous comment, should be 
```
setSlotLength(o2::calibration::INFINITE_TF - 1);
setMaxSlotsDelay(0);
```
to guarantee that the endOfStream call with tf = INFINITE_TF) passes the https://github.com/AliceO2Group/AliceO2/blob/b0f70e90f824acf876f63ca28f8ddeffb6b9523b/Detectors/Calibration/include/DetectorsCalibration/TimeSlotCalibration.h#L315-L317

Reply:
@shahor02 I have to confess that I still do not fully understand the logic.. 
Calling `setMaxSlotsDelay(0)` will set `mFinalizeWhenReady` to `true`. This means that the program will not go through the lines you quote, instead it will reach this portion of code:
https://github.com/AliceO2Group/AliceO2/blob/b0f70e90f824acf876f63ca28f8ddeffb6b9523b/Detectors/Calibration/include/DetectorsCalibration/TimeSlotCalibration.h#L279-L286
In this case, the slot is indeed finalized as soon as the desired statistics is reached, but then the slot is removed and a new one is inserted. However, we do not really need this new slot insertion, because we only use one slot for the full run...

Reply:
> Calling `setMaxSlotsDelay(0)` will set `mFinalizeWhenReady` to `true`. 

No, you are confusing it with `setSlotLength(0)` behaviour,  the mMaxSlotsDelay (the unit is a fraction of the slot length) defines maximum difference `tf_current - slot[I]_end_tf` after which the slot `#I` is allowed to be checked for finalization (i.e. no more TFs contributing to it are expected). 
Though after rechecking the code I see that it will work also w/o `setMaxSlotsDelay(0)` setting, the important part is  `setSlotLength(INFINITE_TF-1)`:
since the endOfStream calls  `checkSlotsToFinalize` with `tf==INFINITE_TF`, and the slot can be finalized only if 
`"current" tf >  slot->getTFEnd()`, the `slot->getTFEnd()` must be `<INFINITE_TF`.

Reply:
@shahor02 my bad, I completely mixed things up! You are absolutely correct, and `setSlotLength(INFINITE_TF-1)` is ok to trigger the slot finalization.

Nevertheless, I would still be in favor of not using the `checkSlotsToFinalize()` function in our code, for the following reasons:
* we initialize the calibrator with `setUpdateAtTheEndOfRunOnly()`, therefore we do not take advantage of the `checkSlotsToFinalize()` call at each TF in the data processing routine
* we anyhow check explicitly the available statistics in the `run()` method (see [here](https://github.com/AliceMCH/AliceO2/blob/9008f679c2ca05695ecbd439dad9d8d6db2ed413/Detectors/MUON/MCH/Calibration/src/BadChannelCalibrationDevice.cxx#L87)) when we do not rely on the EoS
* when the EoS is enabled, we finalize the slot even if the statistics is not enough, because we want to always send the output to the QC for diagnostics purposes - AFAIK this is not possible with the `checkSlotsToFinalize()` method

Do you see any drawbacks?